src/components/pages/SettingsPage.jsx

```diff
@@ -29,34 +29,96 @@ export default class SettingsPage extends SimplePageBase {
         this.state = {
             oldPassword: '',
             newPassword: '',
+            email: this.props.user.getIn(['data', 'email']),
         };
+
     }
 
     renderContent() {
         const passwordPending = this.props.passwordStore.get('isChangePending');
         const passwordChanged = this.props.passwordStore.get('changed');
         const passwordError = this.props.passwordStore.get('changeError');
-        let msg = null;
+        let passwordMsg = null;
 
         if (passwordChanged) {
             if (passwordError) {
-                msg = <p className="SettingsPage-msg-error" id="pages.settings.password.error">Wrong password</p>
+                passwordMsg = <p className="SettingsPage-msg-error" id="pages.settings.password.error">Wrong password</p>
             }
             else {
-                msg = <p className="SettingsPage-msg-success" id="pages.settings.password.success">Password changed</p>
+                passwordMsg = <p className="SettingsPage-msg-success" id="pages.settings.password.success">Password changed</p>
                 setTimeout( () => {
-                    this.props.dispatch(resetPasswordChanged());
+                    this.props.dispatch(resetEmailChanged());
                 }, 10000);
             }
         }
 
-        let submitLabel = this.props.intl.formatMessage(
+        const emailPending = this.props.user.get('isChangePending');
+        const emailChanged = this.props.user.get('changed');
+        const emailError = this.props.user.get('changeError');
+
+        let emailMsg = null;
+
+        if (emailChanged) {
+            if (passwordError) {
+                emailMsg = <p className="SettingsPage-msg-error" id="pages.settings.email.error">Cannot set email</p>
+            } else {
+                emailMsg = <p className="SettingsPage-msg-success" id="pages.settings.email.success">Email changed</p>
+                    setTimeout( () => {
+                        this.props.dispatch(resetEmailChanged());
+                    }, 10000);
```

Comment:
This message displays for 10 seconds, after which it is removed.

At the same time, the standard "E-mail can't be changed" is displayed underneath it. I think that if the e-mail change was successful, we should display the success message without timeout, and we should not display the message that says you can't change your e-mail.

Also, I think we should display the e-mail address itself, so that the user can see that their update was applied.

Otherwise, I foresee users will think that the change failed, and contact us unnecessarily.